test/tests/functional/pbs_soft_walltime.py

```diff
@@ -0,0 +1,634 @@
+        J = Job(TEST_USER, attrs={'Resource_List.soft_walltime': 10})
+        msg = 'Cannot set attribute, read only or insufficient permission'
+
+        jid = None
+        try:
+            jid = self.server.submit(J)
+        except PbsSubmitError as e:
+            self.assertTrue(msg in e.msg[0])
+
+        self.assertEquals(jid, None)
+
+        J = Job(TEST_USER)
+        jid = self.server.submit(J)
+        try:
+            self.server.alterjob(jid, {'Resource_List.soft_walltime': 10},
+                                 runas=TEST_USER)
+        except PbsAlterError as e:
+            self.assertTrue(msg in e.msg[0])
+
+        self.server.expect(JOB, 'Resource_List.soft_walltime',
+                           op=UNSET, id=jid)
+
+    def test_soft_walltime_STF(self):
+        """
+        Test that STF jobs can't have soft_walltime
+        """
+        msg = 'soft_walltime is not supported with Shrink to Fit jobs'
+        J = Job(attrs={'Resource_List.min_walltime': 120, ATTR_h: 'u'})
+        jid = self.server.submit(J)
+        try:
+            self.server.alterjob(jid, {'Resource_List.soft_walltime': 10})
+        except PbsAlterError as e:
+            self.assertTrue(msg in e.msg[0])
+
+        self.server.expect(JOB, 'Resource_List.soft_walltime',
+                           op=UNSET, id=jid)
+
+        J = Job(TEST_USER, attrs={ATTR_h: 'u'})
+        jid = self.server.submit(J)
+        self.server.alterjob(jid, {'Resource_List.soft_walltime': 10})
+        try:
+            self.server.alterjob(jid, {'Resource_List.min_walltime': 120})
+        except PbsAlterError as e:
+            self.assertTrue(msg in e.msg[0])
+
+        self.server.expect(JOB, 'Resource_List.min_walltime',
+                           op=UNSET, id=jid)
+
+        J = Job(TEST_USER, attrs={ATTR_h: 'u'})
+        jid = self.server.submit(J)
+        a = {'Resource_List.soft_walltime': 10,
+             'Resource_List.min_walltime': 120}
+        try:
+            self.server.alterjob(jid, a)
+        except PbsAlterError as e:
+            self.assertTrue(msg in e.msg[0])
+
+        al = ['Resource_List.min_walltime', 'Resource_List.soft_walltime']
+        self.server.expect(JOB, al, op=UNSET, id=jid)
+
+    def test_soft_greater_hard(self):
+        """
+        Test that a job's soft_walltime can't be greater than its hard walltime
+        """
+        msg = 'Illegal attribute or resource value'
+        J = Job(TEST_USER, attrs={'Resource_List.walltime': 120, ATTR_h: 'u'})
+        jid = self.server.submit(J)
+
+        try:
+            self.server.alterjob(jid, {'Resource_List.soft_walltime': 240})
+        except PbsAlterError as e:
+            self.assertTrue(msg in e.msg[0])
+
+        self.server.expect(JOB, 'Resource_List.soft_walltime',
+                           op=UNSET, id=jid)
+
+        J = Job(TEST_USER, {ATTR_h: 'u'})
+        jid = self.server.submit(J)
+        self.server.alterjob(jid, {'Resource_List.soft_walltime': 240})
```

Comment:
why not submit the job with soft_walltime instead of altering it?

Reply:
Job submissions happen with user perms.  Even if the user has manager perms, they still can't submit a job requesting soft_walltime.  I'm not exactly sure for the reason of this.  Maybe we can ask Bob the next time we see him.